SpatialGDK/Source/SpatialGDKTests/SpatialGDK/RPCService/RPCServiceTest.cpp

```diff
@@ -0,0 +1,519 @@
+	{
+		AddEntityComponentToStaticComponentView(StaticComponentView, EntityComponentId, Schema_CreateComponentData(), Authority);
+	}
+
+	Worker_Authority GetClientAuthorityFromRPCEndpointType(ERPCEndpointType RPCEndpointType)
+	{
+		switch (RPCEndpointType)
+		{
+		case CLIENT_AUTH:
+		case SERVER_AND_CLIENT_AUTH:
+			return WORKER_AUTHORITY_AUTHORITATIVE;
+			break;
+		case SERVER_AUTH:
+		default:
+			return WORKER_AUTHORITY_NOT_AUTHORITATIVE;
+			break;
+		}
+	}
+
+	Worker_Authority GetServerAuthorityFromRPCEndpointType(ERPCEndpointType RPCEndpointType)
+	{
+		switch (RPCEndpointType)
+		{
+		case SERVER_AUTH:
+		case SERVER_AND_CLIENT_AUTH:
+			return WORKER_AUTHORITY_AUTHORITATIVE;
+			break;
+		case CLIENT_AUTH:
+		default:
+			return WORKER_AUTHORITY_NOT_AUTHORITATIVE;
+			break;
+		}
+	}
+
+	Worker_Authority GetMultiCastAuthorityFromRPCEndpointType(ERPCEndpointType RPCEndpointType)
+	{
+		return GetServerAuthorityFromRPCEndpointType(RPCEndpointType);
+	}
+
+	void AddEntityToStaticComponentView(USpatialStaticComponentView& StaticComponentView, Worker_EntityId EntityId, ERPCEndpointType RPCEndpointType)
+	{
+		Worker_Authority ClientAuth = GetClientAuthorityFromRPCEndpointType(RPCEndpointType);
+		Worker_Authority ServerAuth = GetServerAuthorityFromRPCEndpointType(RPCEndpointType);
+		Worker_Authority MultiCastAuth = GetMultiCastAuthorityFromRPCEndpointType(RPCEndpointType);
+
+		AddEntityComponentToStaticComponentView(StaticComponentView,
+			SpatialGDK::EntityComponentId(EntityId, SpatialConstants::CLIENT_ENDPOINT_COMPONENT_ID),
+			ClientAuth);
+
+		AddEntityComponentToStaticComponentView(StaticComponentView,
+			SpatialGDK::EntityComponentId(EntityId, SpatialConstants::SERVER_ENDPOINT_COMPONENT_ID),
+			ServerAuth);
+
+		AddEntityComponentToStaticComponentView(StaticComponentView,
+			SpatialGDK::EntityComponentId(EntityId, SpatialConstants::MULTICAST_RPCS_COMPONENT_ID),
+			MultiCastAuth);
+	};
+
+	USpatialStaticComponentView* CreateStaticComponentView(const TArray<Worker_EntityId>& EntityIdArray, ERPCEndpointType RPCEndpointType)
+	{
+		USpatialStaticComponentView* StaticComponentView = NewObject<USpatialStaticComponentView>();
+		for (Worker_EntityId EntityId : EntityIdArray)
+		{
+			AddEntityToStaticComponentView(*StaticComponentView, EntityId, RPCEndpointType);
+		}
+		return StaticComponentView;
+	}
+
+	TUniquePtr<SpatialGDK::SpatialRPCService> CreateRPCService(const TArray<Worker_EntityId>& EntityIdArray,
+		ERPCEndpointType RPCEndpointType,
+		ExtractRPCDelegate RPCDelegate = DefaultRPCDelegate,
+		USpatialStaticComponentView* StaticComponentView = nullptr)
+	{
+		if (StaticComponentView == nullptr)
+		{
+			StaticComponentView = CreateStaticComponentView(EntityIdArray, RPCEndpointType);
+		}
+
+		TUniquePtr<SpatialGDK::SpatialRPCService> RPCService = MakeUnique<SpatialGDK::SpatialRPCService>(RPCDelegate, StaticComponentView);
```

Comment:
Why is this a unique_ptr everywhere? It looks like we just use it within function scope each time.

Reply:
Done